Sources/SwiftProtobuf/BinaryEncodingError.swift

```diff
@@ -13,18 +13,18 @@
 // -----------------------------------------------------------------------------
 
 /// Describes errors that can occur when decoding a message from binary format.
-public enum BinaryEncodingError: Error {
+public enum BinaryEncodingError: Error, Equatable {
```

Comment:
What caused this to be needed?

Reply:
https://github.com/apple/swift-protobuf/pull/1527#discussion_r1440116333 was the reason, though I only see an error when `JSONEncodingError` doesn't conform to `Equatable`. `BinaryEncodingError` could technically stay as-is, but I think this is a reasonable change to have parity